### PR TITLE
fix(export): drop the CoreML torch<2.12 pin and make e2e parity tie-robust

### PR DESCRIPTION
**Pull request by lthphuw**
## Summary

The `[coreml]` extra pinned `torch<2.12` because of a supposed "coremltools MIL lowering regression" that broke `test_outputs_match_pytorch_supervision_image`. That diagnosis was wrong. There is no numeric bug in coremltools, and the pin never fixed anything. This PR removes the pin and makes the parity tests deterministic.

**Root cause:** the two-stage encoder selects its queries with `torch.topk` over 576 token scores. With untrained test weights, neighbouring scores sit about 1e-6 apart, which is the same size as legitimate fp32 rounding differences between eager and the CoreML CPU runtime. When a near-tied pair swaps rank, two reference points trade queries while `query_feat` stays positional, and self-attention spreads the mismatch into every logit (abs diff 0.5–1.5). Repro on torch 2.14, seed 0, `people-walking.jpg`:

| | token 381 | token 128 |
| --- | --- | --- |
| eager | −3.3141088 (rank 208) | −3.3141105 (rank 209) |
| CoreML CPU | −3.3141084 (rank 209) | −3.3141079 (rank 208) |

The two scores are 1.67e-6 apart, and CoreML moves each by less than 1e-6.

**Evidence that it is not coremltools:**
- `ExportedProgram` matches eager bit-exactly (0.0).
- The same `.mlpackage` passes on `CPU_AND_GPU`/`ALL` and fails only on `CPU_ONLY`.
- Dropping any MIL pass, or running a minimal pipeline, changes nothing.
- torch 2.11 fails the same way with seed 7 (ranks 98↔99 and 169↔170).

Which pair flips depends on the seed, input, torch version, compute unit and kernel fusion, which is why the failure looked flaky.

## Changes

- `tests/export/test_coreml_export.py`
  - The e2e fixture exports `num_queries=5`. The top of the ranking is sparse: the smallest top-6 gap is 3.1e-4 across 54 cases, against a worst-case CoreML drift of 1e-5.
  - New `_assert_well_conditioned` precondition (gap ≥ 1e-4) runs before every raw comparison, which stays at 1e-4 on all outputs, masks and keypoints included.
  - New e2e test exporting the shipped 300-query Nano and checking that it runs with eager shapes and finite values.
  - New offline tests with no coremltools dependency, run in the Linux CPU job: helper edge cases, plus the precondition on all three real variants.
  - Removed the incorrect "`np.matmul` overflow / verified-good seed" comment.
- `pyproject.toml`: removed `torch<2.12` and the three `[tool.uv] conflicts` that existed only because of it (`coreml` × `executorch`, `typing`, `ci-executorch-pin`).
- `src/rfdetr/export/_coreml/exporter.py`: rewrote the docstring note to describe the real mechanism.
- `.github/workflows/ci-integrations.yml`: comment only. The job stays on 3.11 because I could not verify 3.10 on macOS.

## Verification

Apple M3 Pro, macOS 26.6.2, coremltools 9.0. Each venv was built fresh with the CI install command (`uv pip install -e ".[coreml,train,augment,cli,visual]" --group tests`) plus `torch==2.x.*`.

```
pytest tests/export/test_coreml_export.py tests/export/test_coreml_op_coverage.py src/rfdetr/export/_coreml/ -m "" -n 1
# torch 2.11.0 / 2.12.1 / 2.14.0: 61 passed each
pytest tests/export/test_coreml_export.py -m e2e_coreml   # with _COREML_EXPORT_SEED = 3 and = 7
# 11 passed on every torch version (develop: 2 failed on torch 2.14, seed 0)
pytest src/ tests/export/ -m "not gpu and not coco17 and not integration and not xla and not tpu"
# 862 passed (develop: 848; +14 = 11 tests + 3 doctests)
uvx uv@0.9.26 lock   # resolves; [coreml] installs torch 2.14.0 + coremltools 9.0 on py3.10–3.14
pre-commit run --all-files
```

- **Mutation-tested:** limiting the margin helper to the first image, dropping the boundary gap, and removing the call-count check fail 1, 8 and 2 of the new tests respectively.
- **Pretrained weights:** raw max-abs diffs are identical on torch 2.11, 2.12 and 2.14 (Nano logits 2.33e-4, SegNano masks 1.33e-3, KeypointPreview keypoints 1.19e-2).
- **Cookbook flow** (`RFDETRSmall`, `dog.jpeg`, threshold 0.5) on all three torch versions:
  - CoreML returns the same 4 detections as eager: dog 0.953, backpack 0.659, person 0.581, car 0.532.
  - With the default `ComputeUnit.ALL`, scores differ by at most 1.1e-4 and boxes by at most 0.005 px.
  - The raw tensors differ by 1.70, so a topk flip happens even with pretrained weights, but it stays below the confidence threshold.

**Not verified:**
- `coreml-parity` (macOS CI) has not run yet. All drift and margin numbers come from this machine.
- The notebook was replayed from its committed code, not executed in place.
- `mypy` fails on `src/rfdetr/models/weights.py:691`, which is pre-existing on `develop`.

## Performance

CI cost only: `e2e_coreml` goes from 10 tests / ~48s to 11 tests / ~65s, because of the extra 300-query Nano export.

## Related issues

Part of #1024 (resolves "root-cause the coremltools MIL numeric divergence and lift the torch<2.12 cap"; no upstream coremltools bug to report).

Follow-up, out of scope: `test_executorch_export.py` says `post_process` scores are "invariant to query-order swaps". They are not; a swap moved them by up to 1.23e-3 here.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -45,6 +45,7 @@ dependencies = [
     "pydantic>=2.0,<3",               # ModelConfig / TrainConfig validation
     "supervision>=0.29.0",       # inference output (Detections, Masks)
     "pyDeprecate>=0.9,<0.10",             # deprecation warnings for legacy APIs; >=0.6 for deprecated_class
+    "ipykernel>=7.2.0",
```
[the remark this answers is not in the document]
Comment (lthphuw) [reply pyproject.toml]: Oh, my bad. I think PyCharm auto-installed ipykernel via uv add when I ran the notebook. Removed in the follow-up.

**Review by Borda — APPROVED** [review]
